### ISSUE TITLE
pix2pix with more layers and p2pHD comparison

### ISSUE THREAD

**Issue by kex243** [junyanz/pytorch-CycleGAN-and-pix2pix#1007]
Hi! Thanks for really great code! I have a question about this implementation. 1) What is the key difference if to compare it to old versions of pix2pix, which had more layers (as I get it to gain more receptive field size). Those versions written years ago, and, after this time, it becomes troublesome to run em not meeting compatibility issues and even after overcoming it, model may collapse to black squares rapidly. IMHO it's caused by old TF versions issues, because I remember running them successfully earlier.

I'm asking because of option called '--n_downsample_global' which is equal to 4 by default with Kernel 3x3. If this is the number of generator layers, it should have much lower effective resolution in comparison to eg 8 layers of usual p2p algorithm. I see additional residual blocks, but do they do all the job? Same for D, which is really low size after epoch saving. Maybe I just dont see some performance pluses of such architecture?

2) Also, question is stupid, but size of the model with default parameters are 700 mb, when 8-layers generator may have up to 3 gb weight on hdd. Does it mean that 8-layers generator stores more information about signs which are not connected by same qualities for network? It is important because of the dataset with great diversity of examples without labels, clean p2p. Running pix2pixHD on the one class for 4 days got me perfect results for 256x256 images with single 128-250 pixels objects, but can I add 2 more classes and work with 512x512 (and after all for 1024x512 with help of convolution) without stucking on timeline? Running such month-long experiment into bad way may discourage anyone.

Maybe I shall use 'n_downsample_global' 5 or 6 for more complicated and bigger size objects and more classes? Will it increase the size of model file (and its informal capacity)? As for now this option just doesn't fit my 8 gb card , but im thinking of upgrading and just don't want to waste it.

Anyway, it's worthless to use splited images, multiple GPU and using batch_norm in first half of the training and then switching to batch 1, even for Windows with minor code changes . Thanks a lot.

Just mentioned that missposted. Questions relate to pix2pixHD.

**Comment by junyanz** [comment]
1. Not sure about the TF version. We only provided PyTorch and Lua versions for pix2pix. 
2. The major difference is coarse-to-fine training in pix2pixHD. Another difference is the loss function (feature matching loss, and VGG loss used in pix2pixHD)
3. Adding two classes is possible. But you need to write your own model loading function that can load part of the layers. 
4. 256x256 to 512x512 is possible. I encourage you to post this question on pix2pixHD repo. 
5. Not sure if increasing `n_downsample_global` will help.  You can also add more ResNet blocks.  If your memory is a concern, you can also reduce `ngf`.

**Comment by kex243** [comment]
Thanks. I will try to get it.
